runtime/lua/vim/lsp/log.lua

```diff
@@ -15,18 +15,53 @@ log.levels = {
   ERROR = 4;
 }
 
+-- FIXME: Hack for docs; these functions get overwritten later
+--- Log a message at level TRACE.
+--@see |vim.lsp.log.debug()| for details
+function log.trace(...) end -- luacheck: no unused args
+-- The description is on this function because it's first alphabetically
+
+--- Log a message at level DEBUG.
+---
+--- When called with arguments, it will log at that level (if applicable, it is
+--- checked either way). Lua tables are converted to strings using
+--- |vim.inspect()|.
+---
+--- When called without arguments, it will check whether the
+--- log level is greater than or equal to this one.
+---
+--- Recommended usage:
+---
+---<pre>
+---    local _ = log.debug() and log.debug("123")
```

Comment:
Can we change this? We should have two examples:

The first is with a static string (like above) then you can do `log.debug("123")`. No need to do the `and` check.

If it's an `expensive_function`, then you can do this strategy (so that Lua does not execute expensive function.

In general, it is not necessary to do the `and` check.

Reply:
Will do. I also forgot documenting params and return values…